draft-irtf-cfrg-hpke.md

```diff
@@ -264,42 +265,52 @@ values are two bytes long.
 
 ## DH-Based KEM
 
-Suppose we are given a Diffie-Hellman group that provides the
-following operations:
+Suppose we are given a Diffie-Hellman group, and a hash function,
+providing the following operations:
 
 - GenerateKeyPair(): Generate an ephemeral key pair `(sk, pk)`
   for the DH group in use
 - DH(sk, pk): Perform a non-interactive DH exchange using the
-  private key sk and public key pk to produce a shared secret
-  of length Npk
+  private key sk and public key pk to produce a Diffie-Hellman
+  shared secret of length Npk
 - Marshal(pk): Produce a fixed-length byte string
   encoding the public key `pk`
 - Unmarshal(enc): Parse a fixed-length byte string to recover a
   public key
+- HashDH(m): Compute the cryptographic hash of input message `m`
+  producing the shared secret of length Nzz returned by the algorithm
+- label_hashdh: The UTF-8 string literal "RFCXXXX DHKEM" of length
```

Comment:
I wonder if it would be simplest to do something like the following

```
def LabeledExtract(salt, label, secret):
    labeledSecret = concat("RFC XXXX", label, secret)  
    return Extract(salt, labeledSecret)
```

This is what TLS 1.3 does to make sure the general label gets added everywhere.

Reply:
Is there a reason to attach the label to the secret vs. the salt?

Reply:
> Is there a reason to attach the label to the secret vs. the salt?

The goal of the label in the end is to separate the input domains of all HMAC calls. This concerns Extract and Expand. The three Expand calls for key, nonce, exporter_secret already have the label as prefix of the second argument, and that's why it's in the second argument for Extract, too (and thus attached to the secret/IKM).

Reply:
> I wonder if it would be simplest to do something like the following […]
> This is what TLS 1.3 does to make sure the general label gets added everywhere.

Ups, I forgot that in my recent update. I'll first write some comments to what I did there, and later try to incorporate this suggestions.